app/Models/Recipe.php

```diff
@@ -2,157 +2,203 @@
+    /**
+     * Filter recipes by categories
+     */
+    public function scopeFilterCategories($query, $requested_categories)
+    {
+        $query->when($requested_categories, function ($query, $categories) {
+            $query->whereHas('categories', function ($query) use ($categories) {
+                $query->whereIn('category_id', $categories);
+            });
+        });
+    }
 
-        })->when($request->ingredients, function ($query, $ingredients) {
+    /**
+     * Filter recipes by ingredients
+     */
+    public function scopeFilterIngredients($query, $requested_ingredients)
+    {
+        $query->when($requested_ingredients, function ($query, $ingredients) {
             $query->whereHas('ingredients', function ($query) use ($ingredients) {
                 $query->whereIn('ingredient_id', $ingredients);
             });
-        })->when($request->collections, function ($query, $collections) {
-            if (Auth::user()) {
+        });
+    }
+
+    /**
+     * Filter recipes by collections
+     */
+    public function scopeFilterCollections($query, $requested_collections)
+    {
+        if (Auth::user()) {
+            $query->when($requested_collections, function ($query, $collections) {
                 $query->whereHas('collections', function ($query) use ($collections) {
                     $query->whereIn('collection_id', $collections);
                 });
-            }
-        })->when($request->categories, function ($query, $categories) {
-            $query->whereHas('categories', function ($query) use ($categories) {
-                $query->whereIn('category_id', $categories);
             });
-        })->when($request->favorites, function ($query, $favorites) {
-            if (Auth::user()) {
-                $favorites === "true" ? $query->where("is_favorite", 1)->where("recipes.user_id", Auth::user()->id) : null;
-            }
-        });
+        }
+    }
+
+    /**
+     * Filter recipes by rating
+     */
+    public function scopeFilterByRating($query, $rating_from, $rating_to)
+    {
+        $query->select('recipes.*', \DB::raw('COALESCE(ROUND(AVG(reviews.rating),2),0) as average_rating'), \DB::raw('COUNT(reviews.id) as review_count'))
+            ->leftJoin('reviews', 'reviews.recipe_id', '=', 'recipes.id')
+            ->groupBy('recipes.id');
+        $from = ($rating_from > 0) && ($rating_from <= 5) ? $rating_from : "0.00";
+        $to = ($rating_to <= 5) && ($rating_to > 0) ? $rating_to : "5.00";
+        $query->havingBetween("average_rating", [$from, $to]);
     }
 
-    /*
+    /**
      * Check if a recipe is public
-     * */
+     */
     public function scopePublic($query)
     {
-        return $query->where("is_public", 1);
+        $query->where("is_public", 1);
     }
 
+    /**
+     * Filter recipes if they are favorite to a user
+     */
+    public function scopeFilterFavorites($query, $is_favorite)
+    {
+        if (Auth::user() && $is_favorite == "true") {
```

Comment:
maybe to use 1 and 0 for $is_favorite instead of "true" and "false". And then you can just put $is_favorite in your if statement